applications/MeshingApplication/python_scripts/mmg_process.py

```diff
@@ -417,12 +422,42 @@ def ExecuteInitializeSolutionStep(self):
                                 self.step = 0  # Reset
                                 self.time = 0.0  # Reset
 
+    def ExecuteFinalize(self):
+        """ This method is executed in order to finalize the simulation and save the refined mesh in a new .mdpa file
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        self.remesh_at_finalize = self.settings["remesh_at_finalize"].GetBool()
+        self.output_final_mesh = self.settings["output_final_mesh"].GetBool()
+        output_mesh_file_name = self.settings["output_mesh_file_name"].GetString()
+        sub_model_part_names_to_remove = self.settings["sub_model_part_names_to_remove"].GetStringArray()
+        if self.remesh_at_finalize:
+            self._ExecuteRefinement()
+            for sub_model_part_name in sub_model_part_names_to_remove:
+                if self.main_model_part.HasSubModelPart(sub_model_part_name):
+                    self.main_model_part.RemoveSubModelPart(sub_model_part_name)
+            if self.output_final_mesh:
+                KratosMultiphysics.ModelPartIO(output_mesh_file_name, KratosMultiphysics.IO.WRITE | KratosMultiphysics.IO.MESH_ONLY).WriteModelPart(self.main_model_part)
```

Comment:
```suggestion
        if self.output_final_mesh:
            KratosMultiphysics.ModelPartIO(output_mesh_file_name, KratosMultiphysics.IO.WRITE | KratosMultiphysics.IO.MESH_ONLY).WriteModelPart(self.main_model_part)
```

Can you put this outside the `self.remesh_at_finalize` condition? In case someone does not remesh at `ExecuteFinalize` but still wants to output the final .mdpa.